src/SprykerEco/Zed/Heidelpay/Business/Encrypter/AesEncrypter.php

```diff
@@ -52,14 +52,19 @@ public function encryptData($data)
     /**
      * @param string $data
      *
-     * @return string
+     * @return string|null
      */
     public function decryptData($data)
     {
         $encryptionKey = $this->config
             ->getEncryptionKey();
 
-        list($encryptedData, $initVector) = explode(static::INIT_VECTOR_SEPARATOR, $data);
+        $dataChunks = explode(static::INIT_VECTOR_SEPARATOR, $data);
+        if (count($dataChunks) !== 2) {
+            return null;
```

Comment:
Would it be better to throw an exception in this case? Or is this ok to happen?

Reply:
Yes, I also had this idea, but sometimes data might be empty or not ready for encryption.